### ISSUE TITLE
Cannot install repo without --user on master

### ISSUE THREAD

**Issue by Jehan** [flatpak/flatpak#514]
So I just built master (current HEAD: commit e38d980) to test commit 84aed61.
It did not see my GNOME platform nor the previously installed gnome repo anymore:

> $ flatpak --user install tutorial-repo org.gimp.Gimp
> Required runtime for org.gimp.Gimp/x86_64/master (org.gnome.Platform/x86_64/3.22) is not installed, searching...
> The required runtime org.gnome.Platform/x86_64/3.22 was not found in a configured remote.
> error: The Application org.gimp.Gimp/x86_64/master requires the runtime org.gnome.Platform/x86_64/3.22 which is not installed
> $ flatpak remote-list
> pitivi        user              
> tutorial-repo user,no-gpg-verify

Well ok… maybe it's just because of the prefix change (previous flatpak was the distrib one). I try to reinstall the remote and sdk:

> $ flatpak remote-add --from gnome https://sdk.gnome.org/gnome.flatpakrepo
> $ flatpak install gnome org.gnome.Platform//3.22 org.gnome.Sdk//3.22
> error: Remote "gnome" not found
> $ flatpak remote-list
> pitivi        user              
> tutorial-repo user,no-gpg-verify

It doesn't work!
If I add the remote as user, it's ok though:

> $ flatpak --user remote-add --from gnome https://sdk.gnome.org/gnome.flatpakrepo
> $ flatpak remote-list 
> gnome         user              
> pitivi        user              
> tutorial-repo user,no-gpg-verify

So why doesn't the remote-add work when I install it system-wide? Or maybe it is installed (there is no error showing after all) but not seen afterwards by remote-list or install? Maybe some hard-coded path vs. custom prefix installation?
Anyway I haven't investigated much. It works with --user at least.

**Comment by alexlarsson** [comment]
If you install with a different prefix you need to use `--with-system-install-dir=/var/lib/flatpak` in configure to see the system installed things from the distribution install (due to different prefix).

The problem you had later is that there is a mismatch in version of flatpak-system-helper and flatpak itself. flatpak-system-helper is dbus-activated, and will launch the distribution install.

If your prefix-built one has the right system-install-dir this should work fine (at least until master has some change to flatpak-system-helper, but thats somewhat uncommon).

**Comment by Jehan** [comment]
> If you install with a different prefix you need to use --with-system-install-dir=/var/lib/flatpak in configure to see the system installed things from the distribution install (due to different prefix).

Yep! Does work. Though just re-configuring, building and installing was not enough (normally it should be). I had to do a make clean before a new build for the option to take effect. Maybe something to look into. Minor inconvenience though.

> If your prefix-built one has the right system-install-dir this should work fine (at least until master has some change to flatpak-system-helper, but thats somewhat uncommon).

It's ok, I don't think there necessarily need to have some exception for this path in flatpak. This kind of prefix issue is actually exactly the kind of problem I expected this to be. I just didn't know what option I had to change (or even if there was one).

Maybe this could be documented somewhere. I don't know. Anyway good on my side. I let you decide if you just close as-is, or if you want to document this a little first (and maybe update the autotools files to rebuild what needs to be when the option change).
